plugins/nf-amazon/src/main/nextflow/cloud/aws/config/AwsS3Legacy.groovy

```diff
@@ -45,6 +45,9 @@ class AwsS3Legacy {
     static protected Map normalizeAwsClientConfig(Map<String,?> client) {
 
         normalizeMemUnit(client, 'uploadChunkSize');
+        normalizeMemUnit(client, 'minimumPartSize');
+        normalizeMemUnit(client, 'multipartThreshold');
+        normalizeMemUnit(client, 'maxNativeMemory');
```

Comment:
Why are these new settings being handled in the legacy config? They should be handled separately, likely `AwsS3Config`

Reply:
it should be considered to remove the [s3Legacy](https://github.com/nextflow-io/nextflow/blob/aws-sdk-v2-fs-impl/plugins/nf-amazon/src/main/nextflow/cloud/aws/config/AwsConfig.groovy#L50-L50) settings

Reply:
I think I would be fine with removing the legacy s3 settings (the `upload*` settings) and using the AWS defaults (or providing some sensible fixed defaults). It just doesn't make sense to expose these settings anymore to the user when there are also new settings like `maxConcurrency` and `maxNativeMemory`

Reply:
In that case let's go ahead and mark these options as "no longer supported" in the docs so as to not confuse users

Reply:
Then, I do not need to change the code in this part, just documentation?

Reply:
Yes

Reply:
Done